From 36a65e1ca575a5db99b51ce38e421b1aa375e172 Mon Sep 17 00:00:00 2001
From: Jesse J. Cook <code.crashenx@gmail.com>
Date: Mon, 19 Mar 2012 15:41:33 -0500
Subject: [PATCH 2/2] Increase NAME_LIST_MAX constants <7084 to 7084

Using uuid strings as names (37 bytes):

e.g.
  "12345678-1234-1234-1234-123456789012\0"

allows 7084 names to be squeezed into a RPC message while supporting
"practical uniqueness" and readability

REMOTE_MESSAGE_PAYLOAD_MAX / 37 = 7084
---
 src/remote/remote_protocol.h |   14 +++++++-------
 src/remote/remote_protocol.x |   14 +++++++-------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/src/remote/remote_protocol.h b/src/remote/remote_protocol.h
index 4ec0895..052f2c4 100644
--- a/src/remote/remote_protocol.h
+++ b/src/remote/remote_protocol.h
@@ -33,19 +33,19 @@ typedef char *remote_nonnull_string;
 
 typedef remote_nonnull_string *remote_string;
 #define REMOTE_DOMAIN_ID_LIST_MAX 16384
-#define REMOTE_DOMAIN_NAME_LIST_MAX 1024
+#define REMOTE_DOMAIN_NAME_LIST_MAX 7084
 #define REMOTE_CPUMAP_MAX 256
 #define REMOTE_VCPUINFO_MAX 2048
 #define REMOTE_CPUMAPS_MAX 16384
 #define REMOTE_MIGRATE_COOKIE_MAX 256
-#define REMOTE_NETWORK_NAME_LIST_MAX 256
-#define REMOTE_INTERFACE_NAME_LIST_MAX 256
-#define REMOTE_DEFINED_INTERFACE_NAME_LIST_MAX 256
-#define REMOTE_STORAGE_POOL_NAME_LIST_MAX 256
-#define REMOTE_STORAGE_VOL_NAME_LIST_MAX 1024
+#define REMOTE_NETWORK_NAME_LIST_MAX 7084
+#define REMOTE_INTERFACE_NAME_LIST_MAX 7084
+#define REMOTE_DEFINED_INTERFACE_NAME_LIST_MAX 7084
+#define REMOTE_STORAGE_POOL_NAME_LIST_MAX 7084
+#define REMOTE_STORAGE_VOL_NAME_LIST_MAX 7084
 #define REMOTE_NODE_DEVICE_NAME_LIST_MAX 16384
 #define REMOTE_NODE_DEVICE_CAPS_LIST_MAX 16384
-#define REMOTE_NWFILTER_NAME_LIST_MAX 1024
+#define REMOTE_NWFILTER_NAME_LIST_MAX 7084
 #define REMOTE_DOMAIN_SCHEDULER_PARAMETERS_MAX 16
 #define REMOTE_NODE_MAX_CELLS 1024
 #define REMOTE_AUTH_SASL_DATA_MAX 65536
diff --git a/src/remote/remote_protocol.x b/src/remote/remote_protocol.x
index 62450c4..75d278e 100644
--- a/src/remote/remote_protocol.x
+++ b/src/remote/remote_protocol.x
@@ -81,7 +81,7 @@ typedef remote_nonnull_string *remote_string;
 const REMOTE_DOMAIN_ID_LIST_MAX = 16384;
 
 /* Upper limit on lists of domain names. */
-const REMOTE_DOMAIN_NAME_LIST_MAX = 1024;
+const REMOTE_DOMAIN_NAME_LIST_MAX = 7084;
 
 /* Upper limit on cpumap (bytes) passed to virDomainPinVcpu. */
 const REMOTE_CPUMAP_MAX = 256;
@@ -96,19 +96,19 @@ const REMOTE_CPUMAPS_MAX = 16384;
 const REMOTE_MIGRATE_COOKIE_MAX = 256;
 
 /* Upper limit on lists of network names. */
-const REMOTE_NETWORK_NAME_LIST_MAX = 256;
+const REMOTE_NETWORK_NAME_LIST_MAX = 7084;
 
 /* Upper limit on lists of interface names. */
-const REMOTE_INTERFACE_NAME_LIST_MAX = 256;
+const REMOTE_INTERFACE_NAME_LIST_MAX = 7084;
 
 /* Upper limit on lists of defined interface names. */
-const REMOTE_DEFINED_INTERFACE_NAME_LIST_MAX = 256;
+const REMOTE_DEFINED_INTERFACE_NAME_LIST_MAX = 7084;
 
 /* Upper limit on lists of storage pool names. */
-const REMOTE_STORAGE_POOL_NAME_LIST_MAX = 256;
+const REMOTE_STORAGE_POOL_NAME_LIST_MAX = 7084;
 
 /* Upper limit on lists of storage vol names. */
-const REMOTE_STORAGE_VOL_NAME_LIST_MAX = 1024;
+const REMOTE_STORAGE_VOL_NAME_LIST_MAX = 7084;
 
 /* Upper limit on lists of node device names. */
 const REMOTE_NODE_DEVICE_NAME_LIST_MAX = 16384;
@@ -117,7 +117,7 @@ const REMOTE_NODE_DEVICE_NAME_LIST_MAX = 16384;
 const REMOTE_NODE_DEVICE_CAPS_LIST_MAX = 16384;
 
 /* Upper limit on lists of network filter names. */
-const REMOTE_NWFILTER_NAME_LIST_MAX = 1024;
+const REMOTE_NWFILTER_NAME_LIST_MAX = 7084;
 
 /* Upper limit on list of scheduler parameters. */
 const REMOTE_DOMAIN_SCHEDULER_PARAMETERS_MAX = 16;
-- 
1.7.2.5

